### DISCUSSION TITLE
How adjustement stripe-webook to edge api routes in nextjs?

### DISCUSSION THREAD

**Discussion by robokonk** [vercel/next.js#40311] [Help]
I want adjustment my stripe-webook to Ede API Routes?
I readed several sources and official documentation:
https://nextjs.org/blog/next-12-2#edge-api-routes-experimental

My code:
https://pastebin.com/vUNdNrjB

I should replace all res to code like this?
```
return new Response(
    JSON.stringify(data, null, 2),
    {
      headers: {
        'content-type': 'application/json;charset=UTF-8',
      }
    }
  )
```

**Comment by icyJoseph** [comment]
Well, I'd be careful migrating to experimental technology, the API might just change.

That being said, I think `micro` might just work on the Edge runtime, but I have doubts about Stripe's package. Because they use `require` and a bunch of Node.js natives that as far as I know are not supported by the edge runtime: https://nextjs.org/docs/api-reference/edge-runtime#unsupported-apis

Otherwise, you've got it right. You gotta replace calls to res, by manually crafted Response instances. 

I think the best way to find out if there's compatibility is to just do it, build and try to run it locally. I think it ought to fail is there's a package that is using unsupported APIs.
